design/construct-tree.md

```diff
@@ -0,0 +1,343 @@
+                    }
+                  ],
+                  "children": [
+                    {
+                      "id": "S3Bucket",
+                      "path": "CdkWorkshopStack/HelloHandler/Code/S3Bucket",
+                      "resourecType": "AWS::S3::Bucket",
+                      "metadata": [
+                        {
+                          "sourceLocation": "sample-source-location"
+                        }
+                      ]
+                    },
+                    {
+                      "id": "S3VersionKey",
+                      "path": "CdkWorkshopStack/HelloHandler/Code/S3VersionKey",
+                      "resourceType": "AWS::S3::Attribute",
+                      "metadata": [
+                        {
+                          "sourceLocation": "sample-source-location"
+                        }
+                      ]
+                    },
+                    {
+                      "id": "AssetBucket",
+                      "path": "CdkWorkshopStack/HelloHandler/Code/AssetBucket",
+                      "resourceType": "AWS::S3::Bucket",
+                      "metadata": [
+                        {
+                          "sourceLocation": "sample-source-location"
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ],
+              "children": [
+                {
+                  "id": "Resource",
+                  "path": "CdkWorkshopStack/HelloHandler/ServiceRole/Resource",
+                  "resourceType": "AWS::IAM::Role",
+                  "metadata": [
+                    {
+                      "sourceLocation": "sample-source-location",
+                      "renderedView": [
+                        {
+                          "logicalId": "HelloHandlerServiceRole11EF7C63"
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Construct metadata
+
+We will also introduce an interface that constructs *can* implement to provide metadata such as properties, additional context, anything a construct would want to report back into the rendered view. Constructs will be able to supply information about themselves for inclusion into the context tree.
+
+The context tree provides the skeleton and the structure for rendering a tree-view of a CDK application. The proposed Interface will be added to `construct.ts` in the core library and implemented by `constructs` that have additional information to contribute to the construct tree :
+
+```typescript
+/***
+ * Displayable information that a construct will contribute towards the context tree*
+ * such as dependencies, properties, links, documentation, etc
+ */
+export interface IDisplayable {
+
+  // id of the construct within the current scope*
+  readonly id: string
```

Comment:
`id` should always be the path of the construct in the tree, this should probably be something like `displayName`